src/Native/System.Net.Http.Native/pal_multi.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_multi.h"
+
+static_assert(PAL_CURLM_OK == CURLM_OK, "");
+static_assert(PAL_CURLM_BAD_HANDLE == CURLM_BAD_HANDLE, "");
+static_assert(PAL_CURLM_BAD_EASY_HANDLE == CURLM_BAD_EASY_HANDLE, "");
+static_assert(PAL_CURLM_OUT_OF_MEMORY == CURLM_OUT_OF_MEMORY, "");
+static_assert(PAL_CURLM_INTERNAL_ERROR == CURLM_INTERNAL_ERROR, "");
+static_assert(PAL_CURLM_BAD_SOCKET == CURLM_BAD_SOCKET, "");
+static_assert(PAL_CURLM_UNKNOWN_OPTION == CURLM_UNKNOWN_OPTION, "");
+static_assert(PAL_CURLM_ADDED_ALREADY == CURLM_ADDED_ALREADY, "");
+
+static_assert(PAL_CURLMSG_DONE == CURLMSG_DONE, "");
+
+extern "C" CURLM* MultiCreate()
+{
+    return curl_multi_init();
+}
+
+extern "C" int32_t MultiDestroy(CURLM* multi_handle)
+{
+    return curl_multi_cleanup(multi_handle);
+}
+
+extern "C" int32_t MultiAddHandle(CURLM* multi_handle, CURL* easy_handle)
+{
+    return curl_multi_add_handle(multi_handle, easy_handle);
+}
+
+extern "C" int32_t MultiRemoveHandle(CURLM* multi_handle, CURL* easy_handle)
+{
+    return curl_multi_remove_handle(multi_handle, easy_handle);
+}
+
+extern "C" int32_t MultiWait(CURLM* multi_handle, int32_t extraFileDescriptor, int32_t* isExtraFileDescriptorActive, int32_t* isTimeout)
+{
+    if (!isExtraFileDescriptorActive || !isTimeout)
```

Comment:
Fine for now, but I think we should make a pass through our shims and replace this kind of argument checking with asserts.  These are really meant to be internal implementation details, as if one managed function was calling another in the same assembly, and in general we don't need to code defensively against other callers.